Dockerfile

```diff
@@ -15,4 +15,8 @@ RUN systemctl set-default multi-user.target
 
 COPY setup /sbin/
 
-CMD ["/sbin/init"]
+STOPSIGNAL SIGRTMIN+3
+VOLUME ["/sys/fs/cgroup"]
+
+# Workaround for docker/docker#27202
+CMD ["/bin/bash", "-c", "exec /sbin/init --log-target=journal 100>&1 1>&-"]
```

Comment:
The docker bug is it closes `/dev/console` early on systemd - so this copies stdout to (arbitrary) fd 100 to prevent it being closed I believe. To be honest I don't fully understand it either, got it from comments at the end of https://github.com/docker/docker/issues/9212

If instead you just use `CMD ["/sbin/init", "--log-target=journal"]` - then everything works but `docker logs` will not show the early boot logs, you can only see logs from after journald starts (via journalctl)